packages/proxy/lib/http/response-middleware.ts

```diff
@@ -334,10 +334,8 @@ const SetInjectionLevel: ResponseMiddleware = function () {
     this.res.setHeader('Origin-Agent-Cluster', '?0')
   }
 
-  this.res.wantsSecurityRemoved = this.config.modifyObstructiveCode && isReqMatchOriginPolicy && (
-    (this.res.wantsInjection === 'full')
-    || resContentTypeIsJavaScript(this.incomingRes)
-  )
+  this.res.wantsSecurityRemoved = this.config.modifyObstructiveCode &&
+    (this.res.wantsInjection === 'full' || this.res.wantsInjection === 'fullCrossOrigin' || (resContentTypeIsJavaScript(this.incomingRes) && this.config.experimentalExpandedModifyObstructiveCode))
```

Comment:
Modify the content if 1st party html full injection, or html that is being injected for `cy.origin`. Currently, we don't have a good mechanism for modifying general 3rd party javascript and has been a considered a bug in the past #8983, so for now we will make sure the `experimentalExpandedModifyObstructiveCode` is enabled before doing so. This is subject to change, and we might even be able to determine if we can modify the code based on the initiator/requestor in the future.

Reply:
I am also wondering if we want to do the same thing for html with this flag. Right now we run into weird edge casesd where HTML is returned from a redirect after the cy.origin block has exited, which turns the rewriting off for that html and framebusts, which is going to make it difficult for users to test

Reply:
I added more aggressive behavior in [5597dfa](https://github.com/cypress-io/cypress/pull/22568/commits/5597dfac8187929947e424655c8bc0e41a0752a3) to remove security for all html/javascript resources, regardless of context, if the flag is enabled. This should prevent the weird gap use case where remote state changes at the end of a `cy.origin` block and html is still being fed to page from that origin, even though we have left that context and the rewriter no longer attempts to rewrite that html.